deployment/deployment.go

```diff
@@ -523,3 +533,97 @@ func meetsAirflowUpgradeReqs(airflowVersion, desiredAirflowVersion string) error
 
 	return nil
 }
+
+// addDagDeploymentArgs adds dag deployment argument to houston request map
+func addDagDeploymentArgs(vars map[string]interface{}, dagDeploymentType, nfsLocation, sshKey, knownHosts, gitRepoURL, gitRevision, gitBranchName, gitDAGDir string, gitSyncInterval int) error {
+	if dagDeploymentType == volumeDeploymentType && nfsLocation != "" {
+		vars["dagDeployment"] = map[string]string{"nfsLocation": nfsLocation, "type": dagDeploymentType}
+	}
+
+	if dagDeploymentType == gitSyncDeploymentType {
+		dagDeploymentConfig := map[string]interface{}{"type": dagDeploymentType}
+		if sshKey != "" && knownHosts != "" {
```

Comment:
what if they only want to pass in the sshkey without the knownHosts?

Reply:
wasn't sure about the internal implementation on the Houston side, but if having just sshKey is a valid input, then can break that logic to:
```
if sshKey != "" {
    <handle ssh key file>
}
if knownHosts != "" {
   <handle known hosts file>
}
```

Reply:
updated